pkg/api/v1/defaults.go

```diff
@@ -187,6 +187,9 @@ func addDefaultingFuncs(scheme *runtime.Scheme) {
 			if obj.Scheme == "" {
 				obj.Scheme = URISchemeHTTP
 			}
+			if obj.HTTPHeaders == nil {
```

Comment:
since it is not a map now, I don't think you need to have a default (a nil slice is appendable, a nil map is useless)